enhancements/evidence-forwarding.md

```diff
@@ -0,0 +1,305 @@
+This works around saving the entire report in Custom Resource Definitions,
+which are bound to limitations of etcd.
+
+Unfortunately, physical volumes have their own short-comings. Reports stored in
+physical volumes still need to be fetched for evidence. Physical volumes can
+also be deleted like any other cluster resource.
+
+A improvement to the idea of putting evidence reports on disk would be to dump
+them into an object or file store. This has an added benefit of allowing
+administrators managing multiple clusters to put evidence in a single location.
+
+## Motivation
+
+The motivation of the enhancement is to store compliance evidence outside the
+cluster and its storage pools.
+
+### Goals
+
+The goal is to provide a way for Kubernetes administrators to configure the
+Compliance Operator to send compliance reports, specifically ARF reports, to an
+external storage pool, like an object or file store.
+
+### Non-Goals
+
+It is not the goal of this enhancement to implement results forwarding, or
+adjust the definition of evidence outside an ARF report.
+
+## Proposal
+
+Provide an additional configuration for scan settings that allows site
+reliability engineers to forward evidence to a storage pool. Today, the
+Compliance Operator provides a `default` `ScanSetting` that integrates with
+persistent storage:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: default
+  namespace: openshift-compliance
+rawResultStorage:
+  pvAccessModes:
+  - ReadWriteOnce
+  rotation: 3
+  size: 1Gi
+roles:
+- worker
+- master
+scanTolerations:
+  default:
+  - operator: Exists
+  schedule: 0 1 * * *
+```
+
+A site reliability engineer could create another `ScanSetting` that bypasses
+storing evidence on physical volumes with the following:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: external-storage
+  namespace: openshift-compliance
+externalResultStorage:
+  endpoint: https://object-storage.example.com/
+  tls:
+roles:
+- worker
+- master
+scanTolerations:
+  default:
+  - operator: Exists
+  schedule: 0 1 * * *
+```
+
+A site reliability engineer could configure both storage mechanisms for redundancy:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: external-storage-and-cluster-storage
+  namespace: openshift-compliance
+rawResultStorage:
+  pvAccessModes:
+  - ReadWriteOnce
+  rotation: 3
+  size: 1Gi
+externalResultStorage:
+  endpoint: https://object-storage.example.com/
+  tls:
+roles:
+- worker
+- master
+scanTolerations:
+  default:
+  - operator: Exists
+  schedule: 0 1 * * *
+```
+
+### User Stories
+
+1. As a site reliability engineer, I want to configure the Compliance Operator
+   to send all ARF reports to a dedicated object store.
+2. As a site reliability engineer, I want to opt out of storing any ARF reports
```

Comment:
I think this is a feature separate from this enhancement.

Reply:
So should we remove it and document that someone can opt-out of storing reports on physical volumes after we implement this?

Reply:
Maybe it's a separate feature, but I think it's worth discussing how it would be configured, since we're touching the storage attributes of `ScanSettings` and we should not change the storage config so that it's difficult to come up with a nice "no storage" option.